### PR TITLE
Support basic PEtab functionality

### PR DESCRIPTION
**Pull request by yannikschaelte**
* Define import functionality.
* Implement import with AMICI model simulations.

### CODE REVIEW

File: doc/examples/petab.ipynb [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+    "# PEtab import"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This illustrates how the PEtab format can be used together with the ODE simulation toolbox AMICI to define ODE based parameter estimation problems for pyABC.\n",
+    "\n",
+    "To use this functionality, you need to have (at least) PEtab and AMICI installed. You can obtain these by installing pyABC with\n",
```
Comment (dweindl) [inline doc/examples/petab.ipynb]: Experience tells, it's good to directly link to the installation instructions ;)

File: doc/examples/petab.ipynb [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+   ],
+   "source": [
+    "model(petab_problem.x_nominal_free_scaled)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "By default, this only returns the log likelihood value. If falso simulated data are to be returned (and stored in the pyABC datastore), pass `store_simulations=True` to the importer."
```
Comment (dweindl) [inline doc/examples/petab.ipynb]:
```suggestion
    "By default, this only returns the log likelihood value. If also simulated data are to be returned (and stored in the pyABC datastore), pass `store_simulations=True` to the importer."
```

File: doc/examples/petab.ipynb [hunk 3]
```diff
@@ -0,0 +1,378 @@
...
...Y5i7bifTl+Ry01ndOKd3B7fTaZbCwoTTurZtdgPVTqxAGBPiCvYd5TfvrOHU5Dbcc+EpbqfTrGWktSN79yH2HSl3OxWfPPnkk45xKxDGhLDKKg93vrmSKo/y7MTBIT2nUmM49jzEitzm0Yv48ssvHeP2r8GYEPbsJ5vJ3L6XxyYMIC2hpdvpNHsDk9sSESZkbmseBeLdd991jFuBMCZEfbmlmL8tyOaKIcmMH9TF7XSCQkxUOP26tAmacQgrEMaEoL2Hy7nrrZV0S2jJH8f1czudoDKkazyr8/ZRXulxO5U6Pf74445xKxDGhKC/L8ym6GAZz04cTMsWNqFCY8pIi6es0sP6HYE/cd+qVasc41YgjAkxRQfLeG3Jdi4b3IX+Xdq4nU7QqbnCXKCbMWOGY9wKhDEhZur/tlBe6eH2c3u6nUpQ6tA6mpR2Mc2iQNTFCoQxIaRm76Gb3bXkNxmp7cjcvhdVdTsVRw8//LBj3K8FQkTGiEiWiGSLyL21xH8lIhtEZI2IfCIiqTVi14nIZu/rOn/maUyosN5D0zgtNZ6ig2XklRx1OxVHWVlZjnG/FQgRCQeeB8YCfYGrRKTvcbutBDJU9VTgHeAv3rbtgIeA04FhwEMiEu+vXI0JBbsPllrvoYl8M3Hf9sCeuG/69OmOcX/2IIYB2aqao6rlwAxgfM0dVHWBqh6bG3cJkOx9fyEwT1VLVHUvMA8Y48dcjQl6UxflWO+hifTqGEdci4hm/zyEPwtEFyCvxud877YT+RlwbHkjn9qKyM0ikikimUVFRSeZrjHBa/fBUqZ/Zb2HphIeJgxOjWdFgBeIBx980DEeEIPUInINkAE8UZ92qjpVVTNUNSMxMdE/yRkTBKYuyqGiSrnDeg9NJiM1nqxdB9l/tMLtVE4oLy/PMe7PAlEApNT4nOzd9h0iMhr4HTBOVcvq09YYU7dveg+Duth8S01oSGo8qrAqb5/bqZzQyy+/7Bj3Z4FYBvQUkW4iEgVMBL6zOoWIDAamUF0cdtcIfQRcICLx3sHpC7zbjDH1NMXbe7j93B5upxJSBqW0JUxgeTNaYe54dRYIEQkTkR/X98CqWglMovoX+0bgbVVdLyKTRWScd7cngFbAv0VklYjM9rYtAR6musgsAyZ7txlj6mH3wVKmL7HegxtatoigT6fWLA/gqb/vu+8+x3idk7CoqkdE
```
Comment (dweindl) [inline doc/examples/petab.ipynb]: ~maybe "parameter_df", since substring "pdf" has different meaning above?~

`pdf` unused? remove.

Comment (dweindl) [inline doc/examples/petab.ipynb]: No refval for `k_exp_hetero` or need to extend plot limits?

File: doc/examples/petab.ipynb [hunk 4]
```diff
@@ -0,0 +1,378 @@
...
+   "metadata": {},
+   "source": [
+    "# PEtab import"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This illustrates how the PEtab format can be used together with the ODE simulation toolbox AMICI to define ODE based parameter estimation problems for pyABC.\n",
```
Comment (dweindl) [inline doc/examples/petab.ipynb]: Introductory sentence on PEtab? Links to AMICI/PEtab github repos?

File: pyabc/petab/amici.py [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+        parameters table).
+    fixed_parameters:
+        Whether to estimate fixed parameters (column ESTIMATE=0 in the
+        parameters table).
+    amici_model:
+        A corresponding compiled AMICI model that allows simulating data for
+        parameters. If not provided, one is created using
+        `amici.petab_import.import_petab_problem`.
+    amici_solver:
+        An AMICI model to simulate the model. If not provided, one is created
```
Comment (dweindl) [inline pyabc/petab/amici.py]:
```suggestion
        An AMICI solver to simulate the model. If not provided, one is created
```

File: pyabc/petab/amici.py [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+            # simulate model
+            sim = simulate_petab(
+                petab_problem=petab_problem,
+                amici_model=amici_model,
+                solver=amici_solver,
+                problem_parameters=par,
+                scaled_parameters=True)
+
+            # return values of interest
+            ret = {'llh': sim['llh']}
```
Comment (dweindl) [inline pyabc/petab/amici.py]: Replace rhs string literal by constants defined in AMICI?

File: pyabc/petab/amici.py [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+            if store_simulations:
+                for i_rdata, rdata in enumerate(ret['rdatas']):
```
Comment (dweindl) [inline pyabc/petab/amici.py]: Replace rhs string literal by constants defined in AMICI?

File: pyabc/petab/amici.py [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+        """
+        Create acceptance kernel.
+
+        Returns
+        -------
+        kernel:
+            A pyabc distribution encoding the kernel function.
+        """
+        def kernel_fun(x, x_0, t, par):
+            # we cheat: the kernel value is computed by the model already
```
Comment (dweindl) [inline pyabc/petab/amici.py]: :fearful:

File: pyabc/petab/base.py [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+                continue
+            if not self.free_parameters and row[petab.C.ESTIMATE] == 1:
+                # ignore free parameters
+                continue
+
+            # pyabc currently only knows objective priors, no
+            #  initialization priors
+            prior_type = row[petab.C.OBJECTIVE_PRIOR_TYPE]
+            pars_str = row[petab.C.OBJECTIVE_PRIOR_PARAMETERS]
+            prior_pars = tuple([float(val) for val in pars_str.split(';')])
```
Comment (dweindl) [inline pyabc/petab/base.py]: might be good to hide the prior parsing inside a PEtab function in the future

File: pyabc/petab/base.py [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+                mean, scale = prior_pars
+                rv = pyabc.RV('laplace', mean, scale)
+            elif prior_type == petab.C.LOG_NORMAL:
+                mean, std = prior_pars
+                rv = pyabc.RV('lognorm', mean, std)
+            elif prior_type == petab.C.LOG_LAPLACE:
+                mean, scale = prior_pars
+                rv = pyabc.RV('loglaplace', mean, scale)
+            else:
+                raise ValueError(f"Cannot handle rior type {prior_type}.")
```
Comment (dweindl) [inline pyabc/petab/base.py]:
```suggestion
                raise ValueError(f"Cannot handle prior type {prior_type}.")
```

File: pyabc/petab/base.py [hunk 11]
```diff
@@ -0,0 +1,142 @@
...
+        Create model. The model takes parameters and simulates data
+        for these. Different model simulation formalisms may be employed.
+
+        .. note::
+            This method must be overwritten in derived classes.
+
+        Returns
+        -------
+        model:
+            Employs some model formalsm to generate simulated data for the
```
Comment (dweindl) [inline pyabc/petab/base.py]:
```suggestion
            Employs some model formalism to generate simulated data for the
```

File: .travis_pip_reqs.txt [hunk 12]
```diff
@@ -30,3 +30,5 @@ flake8
 pytest
 codecov
 pytest-cov
+git+https://github.com/icb-dcm/petab.git@develop
```
[the remark this answers is not in the document]
Comment (yannikschaelte) [reply .travis_pip_reqs.txt]: safer for the moment :)

File: doc/examples/petab.ipynb (hunk 3 above)
Comment (yannikschaelte) [reply doc/examples/petab.ipynb]: yeah it's on the axis exactly

File: pyabc/petab/amici.py (hunk 7 above)
Comment (yannikschaelte) [reply pyabc/petab/amici.py]: what would be the constant's name?

File: pyabc/petab/amici.py (hunk 8 above)
Comment (yannikschaelte) [reply pyabc/petab/amici.py]: can also do it here (have done that actually), but then I would need to really dig into the mappings :laughing:

File: pyabc/petab/amici.py (hunk 7 above)
Comment (dweindl) [reply pyabc/petab/amici.py]: I guess you should know :sweat_smile:  https://github.com/ICB-DCM/AMICI/pull/921/commits/3bac1bbabb785e9168dd0a8f8c993c6591650b43#diff-a1defc0ccc84ee9aa948637c02699a45R18

File: pyabc/petab/amici.py (hunk 8 above)
Comment (dweindl) [reply pyabc/petab/amici.py]: doesn't seem to be necessary ;)

File: pyabc/petab/amici.py (hunk 7 above)
Comment (yannikschaelte) [reply pyabc/petab/amici.py]: uuuups yes of course :rofl:

**Pull request merged**